modules/azure-core-cpp/1.14.1/overlay/BUILD.bazel

```diff
@@ -0,0 +1,120 @@
+load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+bool_flag(
+    name = "build_transport_curl",
+    build_setting_default = True,
+    visibility = ["//visibility:public"],
+)
+
+config_setting(
+    name = "build_transport_curl_setting",
+    flag_values = {":build_transport_curl": "true"},
+)
+
+bool_flag(
+    name = "build_transport_winhttp",
+    build_setting_default = False,
+    visibility = ["//visibility:public"],
+)
+
+config_setting(
+    name = "build_transport_winhttp_setting",
+    flag_values = {":build_transport_winhttp": "true"},
+)
+
+# --@curl//:ssl_lib=openssl must be set as a build flag to use this library if
```

Comment:
You could add a `select` on the config_setting for this flag and without a default branch, then set `no_match_error` to provide a helpful error message in case the user forgets to do this.